src/functions/utils/slackBot.ts

```diff
@@ -20,17 +23,37 @@ export class SlackBot {
 
   private registerCommands(): void {
     const slackBotCommand = process.env.SLACK_BOT_COMMAND as string;
+    const currentStage = process.env.STAGE as string;
+
     this.app.command(slackBotCommand, async ({ command, ack, respond }: SlackCommandMiddlewareArgs) => {
       await ack();
-      await respond(`Hello, <@${command.user_id}>! 👋`);
-
-      console.log("Slack command executed:", {
-        userId: command.user_id,
-        command: slackBotCommand,
-        details: command,
-        timestamp: new Date().toISOString(),
-      });
-      console.log(`Your stage is ${process.env.STAGE}`);
+
+      let userName: string;
+      try {
+        userName = getUserNameForStage(currentStage, command.text, command.user_name);
+
+        const driveItemId = process.env.EXCEL_DRIVE_ITEM_ID as string;
+        const worksheetId = process.env.EXCEL_WORKSHEET_ID as string;
+        const range = "A1:N41";
+
+        console.log(`Fetching data for ${userName}`);
+        const excelData = await fetchExcelDataWithCache(driveItemId, worksheetId, range);
+        const data = findUserRow(excelData, userName);
+
+        await respond(
+          `Koko käytettävissä oleva bonuksesi on: ${data.remainingTotalBudget}€\nRahana maksettava summa: ${data.remainingTotalPayable}€`
+        );
+      } catch (error: unknown) {
+        console.error("Error processing command:", error);
+
+        if (error instanceof Error) {
+          if (error.message === "No username provided in dev mode.") {
+            await respond("No username provided in dev mode. Please provide the username in the command text.");
+          } else {
+            await respond("Your slack username was not found in the bonus excel, contact Mari");
```

Comment:
These are in english but the success response is in finnish :)

Put the texts in a loc file